pkg/controllers/dynakube/oneagent/oneagent_reconciler.go

```diff
@@ -238,9 +240,28 @@ func extractPublicData(dk *dynakube.DynaKube) map[string]string {
 	return data
 }
 
+func (r *Reconciler) calculateTenantTokenHash(ctx context.Context) (string, error) {
+	tenantToken, err := k8ssecret.GetDataFromSecretName(ctx, r.apiReader, types.NamespacedName{
+		Name:      r.dk.OneagentTenantSecret(),
+		Namespace: r.dk.Namespace,
+	}, connectioninfo.TenantTokenKey, log)
+
+	if err != nil {
+		log.Error(err, "secret for tenant token was not available at DaemonSet build time", "dynakube", r.dk.Name)
+		conditions.SetKubeApiError(r.dk.Conditions(), oaConditionType, err)
+	}
```

Comment:
IMHO we should `return err` and do not change the template if K8S API error encountered. We don't know if the secret/tenant has changed or not.